src/libsystemd/sd-event/sd-event.c

```diff
@@ -2318,7 +2318,7 @@ static int source_dispatch(sd_event_source *s) {
 
         if (r < 0)
                 log_debug_errno(r, "Event source %s (type %s) returned error, disabling: %m",
-                                strna(s->description), event_source_type_to_string(s->type));
+                                strna(s->description), strna(event_source_type_to_string(s->type)));
```

Comment:
i changed this bit now, but instead of changing the semantics of s->type I now copy it out early, so that we can always log about it, even if the object is gone. See force pushed patch!